app-backend/tile/src/main/scala/LayerCache.scala

```diff
@@ -53,39 +53,56 @@ object LayerCache extends Config with LazyLogging with KamonTrace {
   private val tileCache = HeapBackedMemcachedClient(memcachedClient)
   private val astCache = HeapBackedMemcachedClient(memcachedClient)
 
-  private val attributeStoreCache: ScaffeineCache[UUID, OptionT[Future, (AttributeStore, Map[String, Int])]] =
+  private val attributeStoreWithMaxZoomsCache: ScaffeineCache[UUID, OptionT[Future, (AttributeStore, Map[String, Int])]] =
     Scaffeine()
       .recordStats()
       .expireAfterAccess(5.minutes)
       .maximumSize(500)
       .build[UUID, OptionT[Future, (AttributeStore, Map[String, Int])]]
 
-  def attributeStoreForLayer(layerId: UUID)(implicit ec: ExecutionContext, projectLayerIds: Set[UUID]): OptionT[Future, (AttributeStore, Map[String, Int])] =
+  private val attributeStoreCache: ScaffeineCache[UUID, OptionT[Future, AttributeStore]] =
+    Scaffeine()
+      .recordStats()
+      .expireAfterAccess(5.minutes)
+      .maximumSize(500)
+      .build[UUID, OptionT[Future, AttributeStore]]
+
+  // TODO: consider having one attribute store per project, layerId should be removed from here
+  def attributeStoreForLayerWithMaxZooms(layerId: UUID)(implicit ec: ExecutionContext, projectLayerIds: Set[UUID]): OptionT[Future, (AttributeStore, Map[String, Int])] =
+    traceName(s"LayerCache.attributeStoreForLayerWithMaxZooms($layerId)") {
+      attributeStoreWithMaxZoomsCache.take(layerId, _ =>
+        traceName(s"LayerCache.attributeStoreForLayerWithMaxZooms($layerId) (no cache)") {
+          val store = PostgresAttributeStore()
+          val maxZooms: Map[String, Int] = store.maxZoomsForLayers(projectLayerIds.map(_.toString))
+          OptionT.fromOption((store, maxZooms).some)
+        }
+      )
+    }
+
+  def attributeStoreForLayer(layerId: UUID)(implicit ec: ExecutionContext): OptionT[Future, AttributeStore] =
     traceName(s"LayerCache.attributeStoreForLayer($layerId)") {
       attributeStoreCache.take(layerId, _ =>
         traceName(s"LayerCache.attributeStoreForLayer($layerId) (no cache)") {
-          val store = PostgresAttributeStore()
-          val maxZooms: Map[String, Int] = blocking { store.maxZoomsForLayers(projectLayerIds.map(_.toString)) }
-          OptionT.fromOption((store, maxZooms).some)
+          OptionT.fromOption(PostgresAttributeStore().some)
```

Comment:
Not sure where to leave this comment, but I'm a little confused about why the logic that used to be in the function was moved to `attributeStoreForLayerWithMaxZooms` and now this one just returns a reference to an attribute store instance?

It seems like places that were using the previous `attributeStoreForLayer` function are still using it the same way.